receiver/k8sobjectsreceiver/README.md

```diff
@@ -0,0 +1,188 @@
+- `mode`: define in which way it collects this type of object, either "poll" or "watch".
+  - `pull` mode will read all objects of this type use the list API at an interval.
+  - `watch` mode will setup a long connection using the watch API to just get updates.
+- `label_selector`: select objects by label(s)
+- `field_selector`: select objects by field(s)
+- `interval`: the interval at which object is pulled, default 60 minutes. Only useful for `pull` mode.
+- `namespaces`: An array of `namespaces` to collect events from. (default = `all`)
+
+
+The full list of settings exposed for this receiver are documented [here](./config.go)
+with detailed sample configurations [here](./testdata/config.yaml).
+
+Follow the below sections to setup various Kubernetes resources required for the deployment.
+
+### Configuration
+
+Create a ConfigMap with the config for `otelcontribcol`. Replace `OTLP_ENDPOINT`
+with valid value.
+
+```bash
+cat <<EOF | kubectl apply -f -
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: otelcontribcol
+  labels:
+    app: otelcontribcol
+data:
+  config.yaml: |
+    receivers:
+      k8sobjects:
+        objects:
+          - name: pods
+            mode: pull
+          - name: events
+            mode: watch
+    exporters:
+      otlp:
+        endpoint: <OTLP_ENDPOINT>
+        tls:
+          insecure: true
+
+    service:
+      pipelines:
+        logs:
+          receivers: [k8sobjects]
+          exporters: [otlp]
+EOF
+```
+
+### Service Account
+
+Create a service account that the collector should use.
+
+```bash
+<<EOF | kubectl apply -f -
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  labels:
+    app: otelcontribcol
+  name: otelcontribcol
+EOF
+```
+
+### RBAC
+
+Use the below commands to create a `ClusterRole` with required permissions and a
+`ClusterRoleBinding` to grant the role to the service account created above.
+Following config will work for collecting pods and events only. You need to add
+appropriate rule for collecting other objects.
+
+```bash
+<<EOF | kubectl apply -f -
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: otelcontribcol
+  labels:
+    app: otelcontribcol
+rules:
+- apiGroups:
+  - ""
+  resources:
+  - events
+  - pods
+  verbs:
+  - get
+  - list
+  - watch
+EOF
+```
+
+```bash
+<<EOF | kubectl apply -f -
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: otelcontribcol
+  labels:
+    app: otelcontribcol
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: otelcontribcol
+subjects:
+- kind: ServiceAccount
+  name: otelcontribcol
+  namespace: default
+EOF
+```
+
+### Deployment
+
+Create a [Deployment](https://kubernetes.io/docs/concepts/workloads/controllers/deployment/) to deploy the collector.
+
+```bash
+<<EOF | kubectl apply -f -
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: otelcontribcol
+  labels:
+    app: otelcontribcol
+spec:
+  replicas: 1
```

Comment:
Please add an important note in the readme that the collector with this receiver must be deployed as one replica, otherwise it'll produce duplicated data.